### PR TITLE
Add 1.3.6 release notes

### PR DESCRIPTION
**Pull request by phestecAMZN**
Signed-off-by: Chris Hesterman <phestech@amazon.com>

### Description
[Describe what this change achieves]

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: release-notes/opensearch-index-management-dashboards-plugin.release-notes-1.3.6.0.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+## Version 1.3.6.0 2022-10-06
+Compatible with OpenSearchDashboards 1.3.0
```
Comment (bowenlan-amzn) [inline release-notes/opensearch-index-management-dashboards-plugin.release-notes-1.3.6.0.md:2]: 1.3.6?

Comment (phestecAMZN) [reply release-notes/opensearch-index-management-dashboards-plugin.release-notes-1.3.6.0.md:2]: That's what they are telling me, see thread in slack

Comment (AWSHurneyt) [inline release-notes/opensearch-index-management-dashboards-plugin.release-notes-1.3.6.0.md:2]: I think we may need to adjust the version on line 2 to `1.3.6`. In the following examples, we adjust the `compatible with...` line to specify the incremental version as well.
https://github.com/opensearch-project/dashboards-reports/blob/1.3/release-notes/opensearch-dashboards-reports.release-notes-1.3.6.0.md
https://github.com/opensearch-project/alerting-dashboards-plugin/blob/main/release-notes/opensearch-alerting-dashboards-plugin.release-notes-2.0.1.0.md?plain=1#L2

**Review by Angie-Zhang — DISMISSED** [review]

File: release-notes/opensearch-index-management-dashboards-plugin.release-notes-1.3.6.0.md (hunk 1 above)
Comment (phestecAMZN) [reply release-notes/opensearch-index-management-dashboards-plugin.release-notes-1.3.6.0.md:2]: got it

**Review by AWSHurneyt — APPROVED** [review]

**Review by Angie-Zhang — APPROVED** [review]

**Pull request merged**